src/Contrib/OtlpHttp/OtlpHttpTransportFactory.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\OtlpHttp;
+
+use InvalidArgumentException;
+use OpenTelemetry\API\Common\Signal\Signals;
+use OpenTelemetry\Contrib\Otlp\OtlpUtil;
+use OpenTelemetry\Contrib\Otlp\Protocols;
+use OpenTelemetry\SDK\Common\Environment\EnvironmentVariablesTrait;
+use OpenTelemetry\SDK\Common\Environment\Variables as Env;
+use OpenTelemetry\SDK\Common\Export\Http\PsrTransport;
+use OpenTelemetry\SDK\Common\Export\Http\PsrTransportFactory;
+use OpenTelemetry\SDK\Common\Export\TransportFactoryInterface;
+use OpenTelemetry\SDK\Common\Otlp\HttpEndpointResolver;
+
+class OtlpHttpTransportFactory implements TransportFactoryInterface
+{
+    use EnvironmentVariablesTrait;
+
+    private const DEFAULT_ENDPOINT = 'http://localhost:4318';
+    private const DEFAULT_COMPRESSION = 'none';
+    private const DEFAULT_PROTOCOL = Protocols::HTTP_PROTOBUF;
+    private const DEFAULT_SIGNAL = Signals::TRACE;
+    private static array $protocols = [
+        Protocols::HTTP_PROTOBUF,
+        Protocols::HTTP_JSON,
+    ];
+    private string $signal = self::DEFAULT_SIGNAL;
+    private string $protocol = self::DEFAULT_PROTOCOL;
+
+    public function create(
+        string $endpoint = null,
+        string $contentType = null,
+        array $headers = [],
+        $compression = null,
+        float $timeout = 10.,
+        int $retryDelay = 100,
+        int $maxRetries = 3,
+        ?string $cacert = null,
+        ?string $cert = null,
+        ?string $key = null
+    ): PsrTransport {
+        $endpoint ??= $this->hasEnvironmentVariable(Env::OTEL_EXPORTER_OTLP_TRACES_ENDPOINT)
```

Comment:
Can we move this into `ExporterFactory::fromEnvironment()` instead?
Should be combined with the `grpc` implementation, the only differences are the default endpoint and the transport factory.

Reply:
Yes, or I might create an otlp-specific exporter factory (which I think can also help to remove some of our sneaky deptrac violations)

Reply:
moved to Otlp\SpanExporterFactory